content/blog/tidymodels-july-2021/index.Rmd

```diff
@@ -0,0 +1,134 @@
+
+We've reorganized the model pages so that there are now sub-packages for each engine. For example, when you use `?linear_reg`, the help page has a _dynamic_ list of engines from parsnip or any parsnip-adjacent package that has been loaded. Here is what the pkgdown site looks like: 
+
+```{r parsnip, echo = FALSE, fig.align='center', out.width="90%"}
+knitr::include_graphics("linear_reg.png")
+```
+
+There is a similar dynamic list in the `See Also` section. 
+
+Each engine page provides basic information about tuning parameters, modes, preprocessing requirements, and anything else that we thing is relevant. For example, for the C5.0 engine for `boost_tree()`:
+
+```{r C50, echo = FALSE, fig.align='center', out.width="90%"}
+knitr::include_graphics("C5.0.png")
+```
+
+Finally, the existing parsnip documentation didn't show the actual fitting and/or prediction in action. A [new pkgdown article](https://parsnip.tidymodels.org/articles/articles/Examples.html) has worked examples demonstrating the use of parsnip models on real data. Here is a screen shot for MARS regression via the earth package: 
+
+```{r earth, echo = FALSE, fig.align='center', out.width="90%"}
+knitr::include_graphics("earth.png")
+```
+
+We think that these changes will greatly improve the whole parsnip experience, especially for new users. 
+
+## Simpler parsnip and workflows interfaces
+
+Our good friend and colleague [David Robinson](https://twitter.com/drob) had [some great ideas](http://varianceexplained.org/r/sliced-ml/#where-tidymodels-can-improve) for specific improvements for our APIs. After some discussion, both of his suggestions were implemented. 
+
+First, we enabled a default engine for parsnip models (you may have noticed this in the screen shots above). This produces simpler code for some model functions and, if a model has a single mode, fitting is as easy as
+
+```r
+# use lm() for regression
+linear_reg() %>% fit(mpg ~ ., data = mtcars)
+```
+
+Another nice feature is more succinct piping for workflows. A preprocessor, such as a formula or recipe, can be piped into `workflow()` now. Also, there is an optional second argument in that function for the model specification. 
+
+Instead of 
+
+```r
+car_rec <- 
+  recipe(mpg ~ ., data = mtcars) %>% 
+  step_ns(disp, deg_free = 5)
+
+car_wflow <- 
+  workflow() %>% 
+  add_recipe(car_rec) %>% 
+  add_model(linear_reg())
+```
+
+you can now use
+
+```r
+car_wflow <- 
+  recipe(mpg ~ ., data = mtcars) %>% 
+  step_ns(disp, deg_free = 5) %>% 
+  workflow(linear_reg()) 
+```  
+
+If you might be on the fence about using tiydmodels, [David's blog post](http://varianceexplained.org/r/sliced-ml/) does an excellent job encapsulating the benefits of our approach, so give it a read. 
+
+
+## Other changes
+
+parsnip now has a generalized additive model function [`gen_additive_mod()`](https://parsnip.tidymodels.org/reference/gen_additive_mod.html)! 
```

Comment:
should this mention where the engines are?

Reply:
Yes. I'll update it with the engine (singular) 😄